### PR TITLE
[MLIR][LLVM] Change addressof builders to use opaque pointers

### PR DESCRIPTION
**Pull request by Dinistro**
This commit changes the builders of the `llvm.mlir.addressof` operations to no longer produce typed pointers.

As a consequence, a GPU to NVVM pattern had to be updated, that still relied on typed pointers.

### CODE REVIEW

File: mlir/examples/toy/Ch6/mlir/LowerToLLVM.cpp [hunk 1]
```diff
@@ -161,10 +161,11 @@ class PrintOpLowering : public ConversionPattern {
     Value globalPtr = builder.create<LLVM::AddressOfOp>(loc, global);
     Value cst0 = builder.create<LLVM::ConstantOp>(loc, builder.getI64Type(),
                                                   builder.getIndexAttr(0));
+    // LLVM::LLVMPointerType::get(IntegerType::get(builder.getContext(), 8)),
```
Comment (gysit) [inline mlir/examples/toy/Ch6/mlir/LowerToLLVM.cpp:164]:
```suggestion
```
nit: looks like some leftover?

**Review by gysit — APPROVED** [review]
LGTM modulo nit!

**Review by ftynse — APPROVED** [review]

File: mlir/examples/toy/Ch6/mlir/LowerToLLVM.cpp (hunk 1 above)
Comment (Dinistro) [reply mlir/examples/toy/Ch6/mlir/LowerToLLVM.cpp:164]: Whups, good catch :)

**Comment by Dinistro** [comment]
PRs cannot be reopened, so I was forced to create a new one: https://github.com/llvm/llvm-project/pull/69292

**Pull request merged**